servicenow/resources/resource_scripted_rest_api.go

```diff
@@ -0,0 +1,174 @@
+	snowClient := serviceNowClient.(client.ServiceNowClient)
+	scriptedRestApi := &client.ScriptedRestApi{}
+	if err := snowClient.GetObject(client.EndpointScriptedRestApi, data.Id(), scriptedRestApi); err != nil {
+		data.SetId("")
+		return err
+	}
+
+	resourceFromScriptedRestApi(data, scriptedRestApi)
+
+	return nil
+}
+
+func createResourceScriptedRestApi(data *schema.ResourceData, serviceNowClient interface{}) error {
+	snowClient := serviceNowClient.(client.ServiceNowClient)
+	scriptedRestApi := resourceToScriptedRestApi(data)
+	if err := snowClient.CreateObject(client.EndpointScriptedRestApi, scriptedRestApi); err != nil {
+		return err
+	}
+
+	resourceFromScriptedRestApi(data, scriptedRestApi)
+
+	return readResourceScriptedRestApi(data, serviceNowClient)
+}
+
+func updateResourceScriptedRestApi(data *schema.ResourceData, serviceNowClient interface{}) error {
+	snowClient := serviceNowClient.(client.ServiceNowClient)
+	if err := snowClient.UpdateObject(client.EndpointScriptedRestApi, resourceToScriptedRestApi(data)); err != nil {
+		return err
+	}
+
+	return readResourceScriptedRestApi(data, serviceNowClient)
+}
+
+func deleteResourceScriptedRestApi(data *schema.ResourceData, serviceNowClient interface{}) error {
+	snowClient := serviceNowClient.(client.ServiceNowClient)
+	return snowClient.DeleteObject(client.EndpointScriptedRestApi, data.Id())
+}
+
+func resourceFromScriptedRestApi(data *schema.ResourceData, scriptedRestApi *client.ScriptedRestApi) {
+	data.SetId(scriptedRestApi.ID)
+	data.Set(scriptedRestApiActive, scriptedRestApi.Active)
+	data.Set(scriptedRestApiConsumes, scriptedRestApi.Consumes)
+	data.Set(scriptedRestApiConsumes_customized, scriptedRestApi.ConsumesCustomized)
+	data.Set(scriptedRestApiEnforce_acl, scriptedRestApi.EnforceACL)
+	data.Set(scriptedRestApiName, scriptedRestApi.Name)
+	data.Set(scriptedRestApiProduces, scriptedRestApi.Produces)
+	data.Set(scriptedRestApiProduces_customized, scriptedRestApi.ProducesCustomized)
+	data.Set(scriptedRestApiServiceId, scriptedRestApi.ServiceId)
+	data.Set(scriptedRestApiBaseURI, scriptedRestApi.BaseURI)
+	data.Set(scriptedRestApiNamespace, scriptedRestApi.Namespace)
+	data.Set(scriptedRestApiDocLink, scriptedRestApi.DocLink)
+	data.Set(scriptedRestApiShortDescription, scriptedRestApi.ShortDescription)
+	data.Set(commonProtectionPolicy, scriptedRestApi.ProtectionPolicy)
+	data.Set(commonScope, scriptedRestApi.Scope)
+}
+
+func resourceToScriptedRestApi(data *schema.ResourceData) *client.ScriptedRestApi {
+	scriptedRestApi := client.ScriptedRestApi{
+		Active:           data.Get(scriptedRestApiActive).(bool),
+		EnforceACL:       data.Get(scriptedRestApiEnforce_acl).(string),
+		Name:             data.Get(scriptedRestApiName).(string),
+		ServiceId:        data.Get(scriptedRestApiServiceId).(string),
+		BaseURI:          data.Get(scriptedRestApiBaseURI).(string),
+		Namespace:        data.Get(scriptedRestApiNamespace).(string),
```

Comment:
Computed values (BaseURI, namespace) should not be in the "To" method since you never "Set" them, you just read them from the service.